control-plane-operator/controllers/hostedcontrolplane/ignitionserver/ignitionserver.go

```diff
@@ -188,6 +189,41 @@ func ReconcileIgnitionServer(ctx context.Context,
 	if hcp.Spec.Platform.Type != hyperv1.IBMCloudPlatform {
 		servingCertSecretName = manifests.IgnitionServerCertSecret("").Name
 	}
+
+	if len(openShiftRegistryOverrides) > 0 {
+		// Ignition server cannot handle ImageContentSourcePolicies or ImageDigestMachineSet, so we need to
+		// fill the registryOverrides in the case of disconnected environments
+		mirrorImage := lookupDisconnectedRegistry(ctx, openShiftRegistryOverrides)
+		if len(mirrorImage) < 1 {
+			// In a weird case where the entry is not in the openShiftRegistryOverrides, we need to figure it out
+			log.Info("failed to find the release entry in the openShiftRegistryOverrides, figuring out the disconnected registry")
+			privateRegistry := strings.Split(hcp.Spec.ReleaseImage, "/")[0]
+			mirrorImage = fmt.Sprintf("%s/openshift/release", privateRegistry)
```

Comment:
Is it possible this could be something else other than `machine-config-operator` or `cluster-config-operator`, do we need an additional if check to report back if it still couldn't figure out the disconnected registry?

Reply:
> Is it possible this could be something else other than machine-config-operator or cluster-config-operator

No, those are the needed ones by the IgnitionServer to generate the Payload. (Tested in IPv6/disconnected env)

> do we need an additional if check to report back if it still couldn't figure out the disconnected registry?

Yeah, makes sense, let me add a new check to report back in case the controller cannot figure the private registry out